src/fromager/context.py

```diff
@@ -64,6 +70,52 @@ def __init__(
             set()
         )
 
+    @classmethod
+    def cpu_count(cls) -> int:
+        """CPU count from scheduler affinity"""
+        return len(os.sched_getaffinity(0))
+
+    @classmethod
+    def available_memory_gib(cls) -> float:
+        """available virtual memory in GiB"""
+        return psutil.virtual_memory().available / (1024**3)
+
+    def parallel_jobs(self, req: Requirement) -> int:
```

Comment:
Does this need to be in the context? Maybe it could standalone in its own module, and then be invoked in wheels.py when the build is started?